pkg/model/iam/types.go

```diff
@@ -29,3 +31,22 @@ func ParseStatements(policy string) ([]*Statement, error) {
 	}
 	return statements, nil
 }
+
+type IAMModelContext struct {
+	// AWSAccountID holds the 12 digit AWS account ID, when running on AWS
+	AWSAccountID string
+
+	Cluster *kops.Cluster
+}
+
+// IAMNameForPodRole determines the name of the IAM Role and Instance Profile to use for the pod role
+func (b *IAMModelContext) IAMNameForPodRole(role PodRole) string {
+	serviceAccount := ServiceAccountForPodRole(role)
+
+	return b.ClusterName() + ".sa." + serviceAccount.Namespace + "." + serviceAccount.Name
```

Comment:
our existing roles are named:
* `masters.$clustername`
* `nodes.$clustername`

so what if we used one of these formats?
* `pods.$clustername.$namespace.$name`
* `serviceaccounts.$clustername.$namespace.$name`
* `sa.$clustername.$namespace.$name`

Roles also support paths which are included in the ARN but are separate fields from the role name itself, so maybe we could use those as well.

Reply:
A good point!  If we're putting the prefix before the cluster name, would `$name.$namespace.sa.$clustername` be a good choice?

Reply:
Regarding naming format, it'd be nice if they sorted such that all of a cluster's roles would grouped together.

We kinda have that currently with our roles beginning with `masters` and `nodes` so that all of the kops roles are near eachother in a list of dozens or hundreds of roles, but adding `dns-controller` or other names would spread them around a bit more.

Just a nit if we have the opportunity to choose, not a big deal either way.